general/optparser.cpp

```diff
@@ -264,7 +265,7 @@ void OptionsParser::ParseCheck(std::ostream &os)
    {
       if (my_rank == 0) { PrintUsage(os); }
 #ifdef MFEM_USE_MPI
-      if (mpi_is_initialized) { MPI_Finalize(); }
+      if (mpi_is_initialized && !Mpi::IsInitialized()) { MPI_Finalize(); }
```

Comment:
This code can also use `MPI_Initialized` and `MPI_Finalized`.

Reply:
It does use `MPI_Initialized`:
```c++
int mpi_err = MPI_Initialized(&mpi_is_initialized);
```

I don't think `MPI_Finalized` will help here

Reply:
We *could* check for `MPI_Finalized` in `Mpi::~Mpi` though, that may be useful.